src/sourmash/commands.py

```diff
@@ -440,15 +460,29 @@ def search(args):
                                              sourmash_args.get_moltype(query))
 
     # downsample if requested
-    if args.scaled:
-        if not query.minhash.scaled:
-            error('cannot downsample a signature not created with --scaled')
+    if isinstance(args.scaled, float):
```

Comment:
I don't understand the purpose of this if statement - what are you checking for? what might args.scaled be if not a float, and how does that relate to the code inside the if statement?

Reply:
this is checking that args.scaled is a float. it may be an int perhaps?

Reply:
The purpose of this statement is to check whether this variable is defined or not (alternative to `if args.scaled`) but with covering for the possibility that `args.scaled` is zero. If this variable always exists, then there's no need for the `isinstance`, and instead, check for `if !args.scaled` or `if args.scaled == 0`.